File: VB/CustomLogonParametersExample.Module/Module.vb
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports System.Reflection
Imports DevExpress.ExpressApp.Security
Imports CustomLogonParametersExample.Module.BusinessObjects
Imports DevExpress.ExpressApp.Security.Strategy


Namespace CustomLogonParametersExample.Module
    Public NotInheritable Partial Class CustomLogonParametersExampleModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Public Overrides Sub Setup(ByVal application As XafApplication)
            AddHandler application.CreateCustomLogonWindowObjectSpace, AddressOf application_CreateCustomLogonWindowObjectSpace
            MyBase.Setup(application)
        End Sub
        Private Sub application_CreateCustomLogonWindowObjectSpace(ByVal sender As Object, ByVal e As CreateCustomLogonWindowObjectSpaceEventArgs)
            Dim objectSpace As IObjectSpace = DirectCast(sender, XafApplication).CreateObjectSpace(GetType(Company))
            CType(e.LogonParameters, CustomLogonParameters).ObjectSpace = objectSpace
            e.ObjectSpace = objectSpace
        End Sub
    End Class
End Namespace

File: CS/CustomLogonParametersExample.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using System.Reflection;
using DevExpress.ExpressApp.Security;
using CustomLogonParametersExample.Module.BusinessObjects;
using DevExpress.ExpressApp.Security.Strategy;


namespace CustomLogonParametersExample.Module {
    public sealed partial class CustomLogonParametersExampleModule : ModuleBase {
        public CustomLogonParametersExampleModule() {
            InitializeComponent();
        }
        public override void Setup(XafApplication application) {
            application.CreateCustomLogonWindowObjectSpace += application_CreateCustomLogonWindowObjectSpace;
            base.Setup(application);
        }
        void application_CreateCustomLogonWindowObjectSpace(
            object sender, CreateCustomLogonWindowObjectSpaceEventArgs e) {
            IObjectSpace objectSpace = ((XafApplication)sender).CreateObjectSpace(typeof(Company));
            ((CustomLogonParameters)e.LogonParameters).ObjectSpace = objectSpace;
            e.ObjectSpace = objectSpace;
        }
    }
}
